### PR TITLE
[HLE/GPU/UI/IR] Replaced push and insert on emplace C++11 and try_emplace C++17

### PR DESCRIPTION
**Pull request by GermanAizek**
@hrydgard, it probably only affects on Debug configuration performance, but it takes up less space if you read.

More info about try_emplace: https://stackoverflow.com/questions/46046828/is-there-any-reason-to-use-stdmapemplace-instead-of-try-emplace-in-c17

### CODE REVIEW

**Comment by Nemoumbra** [comment]
I vote for the `push_back(Object(/* args */))` -> `emplace_back(/* args */)`, but... Why are you trying to introduce `try_emplace` again, after you were told not to in https://github.com/hrydgard/ppsspp/pull/18576?

**Comment by GermanAizek** [comment]
> I vote for the `push_back(Object(/* args */))` -> `emplace_back(/* args */)`, but... Why are you trying to introduce `try_emplace` again, after you were told not to in #18576?

yes, u're right, I probably forgot that I already tried to offer to u.

**Comment by GermanAizek** [comment]
@Nemoumbra fix it https://github.com/hrydgard/ppsspp/pull/18991/commits/b951e7b0b68b6738e2beb38fcc8565e82237ea4a

File: GPU/Common/TextureShaderCommon.cpp [hunk 1]
```diff
@@ -236,7 +236,7 @@ Draw2DPipeline *TextureShaderCache::GetDepalettizeShader(uint32_t clutMode, GETe
 std::vector<std::string> TextureShaderCache::DebugGetShaderIDs(DebugShaderType type) {
 	std::vector<std::string> ids;
 	for (auto &iter : depalCache_) {
-		ids.push_back(StringFromFormat("%08x", iter.first));
```
Comment (Nemoumbra) [inline GPU/Common/TextureShaderCommon.cpp:239]: How does this even compile?? You're pushing a compile-time string literal + the iterator into a vector of `std::strings`!

Comment (GermanAizek) [reply GPU/Common/TextureShaderCommon.cpp:239]: > How does this even compile?? You're pushing a compile-time string literal + the iterator into a vector of `std::strings`!

ha ha ha, you're really right, it's amazing how my compiler didn't mention this, as well as all the others. It's good that all contributors can keep track where and how errors are hidden when reviewing code. It's pity that UB is so impossible to identify.

Comment (Nemoumbra) [reply GPU/Common/TextureShaderCommon.cpp:239]: Ok, why did I say it's the iterator...? It's `iter.first`, not `iter`. That makes more sense - the compiler used some valid `std::string` constructor, probably this one:

image


And no, it's not UB, that's a logic error.

**Comment by hrydgard** [comment]
`emplace_back` is nice for just a little extra performance in some cases, but not really in any of these, it's all very small and simple objects and I think `push_back` with the struct name visible is clearer to read. So I'm choosing not to merge.

**Pull request closed without merging**